services/manifest-repo-export-service/pkg/service/git_test.go

```diff
@@ -0,0 +1,1140 @@
+								ReleaseTrainSource: nil,
+							},
+						},
+					},
+					CommitHash: "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
+				},
+				{
+					Event: api.Event{
+						Uuid:      "00000000-0000-0000-0000-000000000005",
+						CreatedAt: uuid.TimeFromUUID("00000000-0000-0000-0000-000000000005"),
+						EventType: &api.Event_DeploymentEvent{
+							DeploymentEvent: &api.DeploymentEvent{
+								Application:       "app",
+								TargetEnvironment: "staging-1",
+								ReleaseTrainSource: &api.DeploymentEvent_ReleaseTrainSource{
+									UpstreamEnvironment:    "development-1",
+									TargetEnvironmentGroup: conversion.FromString("staging"),
+								},
+							},
+						},
+					},
+					CommitHash: "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
+				},
+			},
+			expectedResponse: &api.GetCommitInfoResponse{
+				CommitHash:    "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
+				LoadMore:      false,
+				CommitMessage: "",
+				TouchedApps: []string{
+					"app",
+				},
+				Events: []*api.Event{
+					{
+						Uuid:      "00000000-0000-0000-0000-000000000002",
+						CreatedAt: uuid.TimeFromUUID("00000000-0000-0000-0000-000000000002"),
+						EventType: &api.Event_CreateReleaseEvent{
+							CreateReleaseEvent: &api.CreateReleaseEvent{
+								EnvironmentNames: []string{
+									"development-1",
+									"staging-1",
+								},
+							},
+						},
+					},
+					{
+						Uuid:      "00000000-0000-0000-0000-000000000003",
+						CreatedAt: uuid.TimeFromUUID("00000000-0000-0000-0000-000000000003"),
+						EventType: &api.Event_DeploymentEvent{
+							DeploymentEvent: &api.DeploymentEvent{
+								Application:        "app",
+								TargetEnvironment:  "development-1",
+								ReleaseTrainSource: nil,
+							},
+						},
+					},
+
+					{
+						Uuid:      "00000000-0000-0000-0000-000000000005",
+						CreatedAt: uuid.TimeFromUUID("00000000-0000-0000-0000-000000000005"),
+						EventType: &api.Event_DeploymentEvent{
+							DeploymentEvent: &api.DeploymentEvent{
+								Application:        "app",
+								TargetEnvironment:  "staging-1",
+								ReleaseTrainSource: nil,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			repo, _ := setupRepositoryTestWithPath(t)
+
+			uuidGenerate := testutil.NewIncrementalUUIDGenerator()
+			pageSize := 100
+			if tc.testPageSize {
+				uuidGenerate = testutil.NewIncrementalUUIDGeneratorForPageSizeTest()
+				pageSize = 2
+			}
+			ctx := rp.AddGeneratorToContext(testutil.MakeTestContext(), uuidGenerate)
+			dbHandler := repo.State().DBHandler
+
+			err := dbHandler.WithTransactionR(ctx, 0, false, func(ctx context.Context, transaction *sql.Tx) error {
+				err := dbHandler.DBWriteMigrationsTransformer(ctx, transaction)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBInsertApplication(ctx, transaction, "app", 1, db.AppStateChangeCreate, db.DBAppMetaData{Team: "team"})
```

Comment:
this test is very big. Is there a way to split it up?

Reply:
Moved the DB setup parts to a new function now.
The rest is the same as the cd service